src/DIRAC/WorkloadManagementSystem/OptimizerAdministrator/OptimizerAdministrator.py

```diff
@@ -0,0 +1,88 @@
+"""
+The OptimizerAdministrator determines the chain of Optimizing Agents
+that must work on the job prior to the scheduling decision and
+launches thoses agents
+
+Initially this takes jobs in the received state and starts the jobs on the
+optimizer chain.
+It ends by setting the job in the Waiting state
+"""
+from DIRAC import S_OK
+from DIRAC.Core.Utilities.ObjectLoader import ObjectLoader
+from DIRAC.WorkloadManagementSystem.Client import JobMinorStatus, JobStatus
+from DIRAC.WorkloadManagementSystem.Client.JobState.JobState import JobState
+from DIRAC.WorkloadManagementSystem.OptimizerAdministrator.OptimizerCaller import OptimizerCaller
+
+
+# TODO: Every method here could become a function, should I do it ?
+# Should I set jobState as an attribute of OptimizerAdministrator ?
```

Comment:
Would that be useful?

Reply:
I ended up setting jobState as an attribute of the OptimizerAdminstrator  constructor for logging reasons.  It was better to instantiate the logger in the  OptimizerAdministrator constructor, and since the logger requires the job id, then jobState had to be passed to the OptimizerAdministrator constructor.